python/hail/expr/expression.py

```diff
@@ -806,175 +1218,545 @@ def __radd__(self, other):
         return self._method("contains", TBoolean(), locus)
 
     @property
     def end(self):
+        """Returns the end locus.
+
+        Returns
+        -------
+        :class:`LocusExpression`
+            End locus.
+        """
         return self._field("end", TLocus())
 
     @property
     def start(self):
+        """Returns the start locus.
+
+        Returns
+        -------
+        :class:`LocusExpression`
+            Start locus.
+        """
         return self._field("start", TLocus())
 
 
 class AltAlleleExpression(Expression):
+    """Expression of type :class:`hail.expr.types.TAltAllele`."""
     @property
     def alt(self):
+        """Returns the alternate allele string.
+
+        Returns
+        -------
+        :class:`StringExpression`
+        """
         return self._field("alt", TString())
 
+    @property
+    def ref(self):
+        """Returns the reference allele string.
+
+        Returns
+        -------
+        :class:`StringExpression`
+        """
+        return self._field("ref", TString())
+
     def category(self):
+        """Returns the string representation of the alternate allele class.
+
+        Returns
+        -------
+        :class:`StringExpression`
+            One of ``"SNP"``, ``"MNP"``, ``"Insertion"``, ``"Deletion"``, ``"Star"``, ``"Complex"``.
+        """
         return self._method("category", TString())
 
     def is_complex(self):
+        """Returns true if the polymorphism is not a SNP, MNP, star, insertion, or deletion.
+
+        Returns
+        -------
+        :class:`BooleanExpression`
+        """
         return self._method("isComplex", TBoolean())
 
     def is_deletion(self):
+        """Returns true if the polymorphism is a deletion.
+
+        Returns
+        -------
+        :class:`BooleanExpression`
+        """
         return self._method("isDeletion", TBoolean())
 
     def is_indel(self):
+        """Returns true if the polymorphism is an insertion or deletion.
+
+        Returns
+        -------
+        :class:`BooleanExpression`
+        """
         return self._method("isIndel", TBoolean())
 
     def is_insertion(self):
+        """Returns true if the polymorphism is an insertion.
+
+        Returns
+        -------
+        :class:`BooleanExpression`
+        """
         return self._method("isInsertion", TBoolean())
 
     def is_mnp(self):
+        """Returns true if the polymorphism is a MNP.
+
+        Returns
+        -------
+        :class:`BooleanExpression`
+        """
         return self._method("isMNP", TBoolean())
 
     def is_snp(self):
+        """Returns true if the polymorphism is a SNP.
+
+        Returns
+        -------
+        :class:`BooleanExpression`
+        """
         return self._method("isSNP", TBoolean())
 
     def is_star(self):
+        """Returns true if the polymorphism is not a star (upstream deletion) allele.
```

Comment:
Is that right? `is_star` returns true if not a star allele?

Reply:
will fix.